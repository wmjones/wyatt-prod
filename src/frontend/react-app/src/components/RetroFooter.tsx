import React from 'react';

const RetroFooter: React.FC = () => {
  return (
    <footer className="mt-20 text-center">
      <div className="zigzag mb-8"></div>

      <div className="grid grid-cols-1 md:grid-cols-3 gap-8 mb-8">
        <div>
          <h4 className="retro-text text-xl text-retro-teal mb-4">NAVIGATION</h4>
          <ul className="space-y-2 mono-text">
            <li><a href="/" className="hover:text-retro-purple">Home</a></li>
            <li><a href="/demo" className="hover:text-retro-purple">Demo</a></li>
            <li><a href="/about" className="hover:text-retro-purple">About</a></li>
          </ul>
        </div>

        <div>
          <h4 className="retro-text text-xl text-retro-teal mb-4">RESOURCES</h4>
          <ul className="space-y-2 mono-text">
            <li><a href="https://d3js.org/" target="_blank" rel="noopener noreferrer" className="hover:text-retro-purple">D3.js</a></li>
            <li><a href="https://reactjs.org/" target="_blank" rel="noopener noreferrer" className="hover:text-retro-purple">React</a></li>
            <li><a href="https://aws.amazon.com/" target="_blank" rel="noopener noreferrer" className="hover:text-retro-purple">AWS</a></li>
          </ul>
        </div>

        <div>
          <h4 className="retro-text text-xl text-retro-teal mb-4">CONNECT</h4>
          <p className="mono-text mb-2">Join our retro visualization community!</p>
          <div className="flex justify-center space-x-4">
            <a href="mailto:contact@example.com" aria-label="Email" className="text-retro-purple hover:text-retro-pink text-xl">
              <span>ðŸ“§</span>
            </a>
            <a href="https://example.com/chat" aria-label="Chat" className="text-retro-purple hover:text-retro-pink text-xl">
              <span>ðŸ’¬</span>
            </a>
            <a href="tel:+12345678900" aria-label="Phone" className="text-retro-purple hover:text-retro-pink text-xl">
              <span>ðŸ“±</span>
            </a>
          </div>
        </div>
      </div>

      <p className="mono-text text-sm">
        D3 Visualization Dashboard - Â© {new Date().getFullYear()} - Deployed via CI/CD Pipeline
      </p>
      <p className="mono-text text-xs mt-2 text-gray-500">
        <span className="animate-blink">â–º</span> Best viewed with Netscape Navigator <span className="animate-blink">â—„</span>
      </p>
    </footer>
  );
};

export default RetroFooter;
