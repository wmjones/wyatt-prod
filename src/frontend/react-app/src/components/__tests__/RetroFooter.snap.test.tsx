import React from 'react';
import renderer from 'react-test-renderer';
import RetroFooter from '../RetroFooter';

describe('RetroFooter Snapshots', () => {
  // Mock current year to ensure snapshot doesn't change every year
  const originalDate = global.Date;

  beforeEach(() => {
    // Mock Date to return fixed year
    global.Date = class extends Date {
      constructor(date) {
        if (date) {
          return super(date);
        }
        return new originalDate('2025-01-01T00:00:00.000Z');
      }
      getFullYear() {
        return 2025;
      }
    };
  });

  afterEach(() => {
    // Restore original Date
    global.Date = originalDate;
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<RetroFooter />)
      .toJSON();
    // Match against the existing snapshot
    expect(tree).toMatchSnapshot();
  });
});
