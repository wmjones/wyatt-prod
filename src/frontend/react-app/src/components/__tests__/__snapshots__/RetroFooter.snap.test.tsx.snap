// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RetroFooter Snapshots renders correctly 1`] = `
<footer
  className="mt-20 text-center"
>
  <div
    className="zigzag mb-8"
  />
  <div
    className="grid grid-cols-1 md:grid-cols-3 gap-8 mb-8"
  >
    <div>
      <h4
        className="retro-text text-xl text-retro-teal mb-4"
      >
        NAVIGATION
      </h4>
      <ul
        className="space-y-2 mono-text"
      >
        <li>
          <a
            className="hover:text-retro-purple"
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            className="hover:text-retro-purple"
            href="/demo"
          >
            Demo
          </a>
        </li>
        <li>
          <a
            className="hover:text-retro-purple"
            href="/about"
          >
            About
          </a>
        </li>
      </ul>
    </div>
    <div>
      <h4
        className="retro-text text-xl text-retro-teal mb-4"
      >
        RESOURCES
      </h4>
      <ul
        className="space-y-2 mono-text"
      >
        <li>
          <a
            className="hover:text-retro-purple"
            href="https://d3js.org/"
            rel="noopener noreferrer"
            target="_blank"
          >
            D3.js
          </a>
        </li>
        <li>
          <a
            className="hover:text-retro-purple"
            href="https://reactjs.org/"
            rel="noopener noreferrer"
            target="_blank"
          >
            React
          </a>
        </li>
        <li>
          <a
            className="hover:text-retro-purple"
            href="https://aws.amazon.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            AWS
          </a>
        </li>
      </ul>
    </div>
    <div>
      <h4
        className="retro-text text-xl text-retro-teal mb-4"
      >
        CONNECT
      </h4>
      <p
        className="mono-text mb-2"
      >
        Join our retro visualization community!
      </p>
      <div
        className="flex justify-center space-x-4"
      >
        <a
          aria-label="Email"
          className="text-retro-purple hover:text-retro-pink text-xl"
          href="mailto:contact@example.com"
        >
          <span>
            ðŸ“§
          </span>
        </a>
        <a
          aria-label="Chat"
          className="text-retro-purple hover:text-retro-pink text-xl"
          href="https://example.com/chat"
        >
          <span>
            ðŸ’¬
          </span>
        </a>
        <a
          aria-label="Phone"
          className="text-retro-purple hover:text-retro-pink text-xl"
          href="tel:+12345678900"
        >
          <span>
            ðŸ“±
          </span>
        </a>
      </div>
    </div>
  </div>
  <p
    className="mono-text text-sm"
  >
    D3 Visualization Dashboard - Â© 
    2025
     - Deployed via CI/CD Pipeline
  </p>
  <p
    className="mono-text text-xs mt-2 text-gray-500"
  >
    <span
      className="animate-blink"
    >
      â–º
    </span>
     Best viewed with Netscape Navigator 
    <span
      className="animate-blink"
    >
      â—„
    </span>
  </p>
</footer>
`;
